packages/app/obojobo-document-engine/src/scripts/common/registry.js

```diff
@@ -40,34 +40,29 @@ class _Registry {
 		return this
 	}
 
+	cloneBlankNode(templateObject) {
+		return JSON.parse(JSON.stringify(templateObject))
+	}
+
 	registerModel(className, opts = {}) {
 		const item = items.get(className)
 		if (item) opts = Object.assign(opts, item)
 
-		items.set(className, opts)
-
 		opts = Object.assign(
 			{
 				type: null,
 				default: false,
-				insertItem: null,
-				componentClass: null,
-				commandHandler: null,
 				variables: {},
-				init() {},
-				// editor
-				// pull out to editor registry eventually
-				name: '',
-				icon: null,
-				isInsertable: false,
-				slateToObo: null,
-				oboToSlate: null,
-				plugins: null,
-				supportsChildren: false
+				templateObject: '',
+				init() {}
```

Comment:
The removed properties seem useful to ensure that a node has all of these in their opts config. I'm sure I'm missing something but why were these removed?

Reply:
https://github.com/ucfopen/Obojobo/blob/35d5eee807ed08cf36ac02a808876bcf2ef6bf18/packages/app/obojobo-document-engine/src/scripts/common/registry.js#L49

Note that `opts` is passed into `items.set`, THEN `opts` is re-assigned a new value (where it gets these default properties).  At this point opts will be a different object than the one sent to `items`.

I don' t think this is the intended purpose AND it proves those properties were never used.

Reply:
Ahh, makes sense to me now